### PR TITLE
Add stop start functionality to dataproc operators

### PR DESCRIPTION
**Pull request by molcay**
Implement DataprocStartClusterOperator and DataprocStopClusterOperator
Also, Add the following logic to DataprocCreateClusterOperator:
  - if the cluster already exists, and it is in stopped state, then start the cluster.
 

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/operators/dataproc.py [hunk 1]
```diff
@@ -1057,6 +1071,196 @@ def _delete_cluster(self, hook: DataprocHook):
...
+        This method's return value will be also return from the execute function.
+        This method returns None as default, if you want to return something else, then you can override it.
+        """
+
+        return None
+
+    def execute(self, context: Context):
+        hook = DataprocHook(gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain)
+        cluster: Cluster = self._get_cluster(hook)
+        print("CLUSTER", cluster)
```
Comment (MaksYermak) [inline airflow/providers/google/cloud/operators/dataproc.py:1189]: Could you change this row from `print()` to `self.log.info` or remove if this row was added for test purpose?

File: airflow/providers/google/cloud/operators/dataproc.py [hunk 2]
```diff
@@ -1057,6 +1071,196 @@ def _delete_cluster(self, hook: DataprocHook):
...
+            region=self.region,
+            project_id=self.project_id,
+            cluster_name=self.cluster_name,
+            cluster_uuid=self.cluster_uuid,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+
+    def _get_return_value(self, hook: DataprocHook):
```
Comment (MaksYermak) [inline airflow/providers/google/cloud/operators/dataproc.py:1234]: Do we strongly need this function? Does the `wait_for_operation` function not return `operation.result` with cluster info?

File: airflow/providers/google/cloud/operators/dataproc.py (hunk 1 above)
Comment (molcay) [reply airflow/providers/google/cloud/operators/dataproc.py:1189]: it was just for test purposes, I will remove the line.

File: airflow/providers/google/cloud/operators/dataproc.py (hunk 2 above)
Comment (molcay) [reply airflow/providers/google/cloud/operators/dataproc.py:1234]: I tested it locally, and it seems we can just return `Cluster.to_dict(result)`  and remove this method.
At first, I was thinking that we can return Cluster information for the start operator but it is not relevant to stop operator, however returning cluster information can be useful for both cases.

**Review by MaksYermak — APPROVED** [review]

**Pull request closed without merging**